sklearn/meson.build

```diff
@@ -18,6 +18,38 @@ if is_mingw
   add_project_arguments('-mlong-double-64', language: 'c')
 endif
 
+if not py.version().version_compare('>=3.9')
+  error('scikit-learn requires Python>=3.9, got ' + py.version() + ' instead')
+endif
+cython_min_version = run_command(py, ['_min_dependencies.py', 'cython'], check: true).stdout().strip()
+if not cython.version().version_compare('>=' + cython_min_version)
+  error('scikit-learn requires Cython>=' + cython_min_version + ', got ' + cython.version() + ' instead')
+endif
```

Comment:
I was indeed able to build with cython 3.0.8 without any complain. Since the min dependencies are listed in ``pyproject.toml``, is there no automated way to make meson aware of them ?

Reply:
I haven't found anything, pyproject.toml is Python-specific and Meson is a generic build tool. You can run a command from `meson.build` and check its exit code, see https://mesonbuild.com/External-commands.html for more details.

I think there are a few options:
1. do what is currently done in this PR, `_min_dependencies.py` for getting minimum versions + Meson-native version comparison. I am not saying this is great, but to me this seems a reasonable trade-off. I spent a bit of time trying to find something that I found significantly better and I was not able to, but better suggestions welcome!
2. write an external Python script but we need `packaging` to compare versions. `packaging` is vendored inside pip, maybe we could assume that in most cases you have pip when you are trying to build scikit-learn ...
3. use our vendored `packaging` but we need `__SKLEARN_SETUP__` hack + `sys.path` manipulations to import from `sklearn` before `sklearn` is built
4. maybe better suggestions?

Note: we could give a more complicated `pip` install command with `--check-build-dependencies` but that would only work when building with Meson through pip. If one day we use `spin`, `spin build` will call meson directly (i.e. not going through pip) and as such will not error in there are some build dependencies missing.

One thing I had some hope for would be to use `run_command` + right `pip` incantation to check build-dependencies (assuming pip is installed which is the most common case I think). I thought maybe
```
pip install . --dry-run --no-build-isolation --check-build-dependencies`
``

may work, but no it does not, causing an error because scikit-learn is already being built
```
LookupError: file:///home/lesteve/dev/scikit-learn is already being built: file:///home/lesteve/dev/scikit-learn
````

Reply:
> do what is currently done in this PR, _min_dependencies.py for getting minimum versions + Meson-native version comparison. I am not saying this is great, but to me this seems a reasonable trade-off. I spent a bit of time trying to find something that I found significantly better and I was not able to, but better suggestions welcome!

I think that this approach is reasonable in that the benefit outweighs the complexity, however I'm unfamiliar with the meson build system so, similarly, I can't say for sure that there isn't a better way of doing it. Thanks for looking into this!